cpp/CMakeLists.txt

```diff
@@ -579,15 +579,22 @@ if(BUILD_SHARED_LIBS)
     )
 
     if(BUILD_MG_ALGOS)
-      set(CUVS_COMMS_DEPENDENCY nccl)
+      rapids_find_generate_module(
+        NCCL
+        HEADER_NAMES nccl.h
+        LIBRARY_NAMES nccl
+      )
+      find_package(NCCL REQUIRED)
+      target_link_libraries(cuvs_objs PRIVATE NCCL::NCCL)
+      target_link_libraries(cuvs PUBLIC NCCL::NCCL)
```

Comment:
I believe this should be a `PRIVATE` link. @viclafargue correct me if I am wrong but does any cuvs header include NCCL header or a RAFT header that includes NCCL header?

Reply:
Thanks for noticing this. The public cuVS MNMG API should only take in a `raft::resources`, everything related to NCCL is hidden inside the implementation.

Reply:
What is currently the status though? Are we exposing the NCCL header directly or indirectly to cuVS users from a header file?

Reply:
re-testing localled shows that I can have this as PRIVATE.